src/argilla/client/client.py

```diff
@@ -583,18 +587,39 @@ def compute_metric(
     def add_dataset_labeling_rules(self, dataset: str, rules: List[LabelingRule]):
         """Adds the dataset labeling rules"""
         for rule in rules:
-            text_classification_api.add_dataset_labeling_rule(
-                self._client,
-                name=dataset,
-                rule=rule,
-            )
-
-    def update_dataset_labeling_rules(self, dataset: str, rules: List[LabelingRule]):
+            try:
+                text_classification_api.add_dataset_labeling_rule(
+                    self._client,
+                    name=dataset,
+                    rule=rule,
+                )
+            except AlreadyExistsApiError:
+                _LOGGER.warning(
+                    f"Rule {rule} already exists. Please, update the rule instead."
+                )
+            except Exception as ex:
+                _LOGGER.warning(f"Cannot create rule {rule}: {ex}")
+
+    def update_dataset_labeling_rules(
+        self,
+        dataset: str,
+        rules: List[LabelingRule],
+    ):
         """Updates the dataset labeling rules"""
         for rule in rules:
-            text_classification_api.update_dataset_labeling_rule(
-                self._client, name=dataset, rule=rule
-            )
+            try:
+                text_classification_api.update_dataset_labeling_rule(
+                    self._client,
+                    name=dataset,
+                    rule=rule,
+                )
+            except NotFoundApiError:
+                _LOGGER.info(f"Rule {rule} does not exists, creating...")
+                text_classification_api.add_dataset_labeling_rule(
+                    self._client, name=dataset, rule=rule
+                )
+            except Exception as ex:
+                _LOGGER.warning(f"Cannot update rule {rule}: {ex}")
 
     def delete_dataset_labeling_rules(self, dataset: str, rules: List[LabelingRule]):
```

Comment:
Do we want to add exceptions here?

Reply:
delete_dataset_labeling_rules, fetch_dataset_labeling_rules and rule_metrics_for_dataset

Reply:
The more conflictive flow is related to adding/updating rules. So I would like to keep the "delete" part as is. We can see later if we should also add error handling to partially delete a list of rules.

Reply:
```suggestion
    def delete_dataset_labeling_rules(self, dataset: str, rules: List[LabelingRule]):
        for rule in rules:
                try:
                    text_classification_api.delete_dataset_labeling_rule(
                        self._client, name=dataset, rule=rule
                    )
                except Exception as ex:
                    _LOGGER.warning(f"Cannot delete rule {rule}: {ex}")
```